client/my-sites/checkout/src/components/checkout-main.tsx

```diff
@@ -224,6 +226,24 @@ export default function CheckoutMain( {
 		addProductsToCart,
 	} = useShoppingCart( cartKey );
 
+	const { shouldReplacePlan, getReplacedPlanSlug } = useFCCARestrictions();
+
+	useEffect( () => {
```

Comment:
🤔 To me this logic feels like it belongs inside the shopping cart on the backend rather than in the frontend code here. The shopping cart already uses lots of logic to determine which products should be added, removed, or replaced given various criteria. It also provides unit tests to guarantee that the behavior works and is more reliable than hoping the client will do the right thing. Also on the frontend there's no guarantee that some other checkout (eg: one-click checkout) might bypass this effect.

Reply:
Those are fair points. Thank you!
I moved the cart item replace functionality to the endpoint in D149866-code.

Reply:
Thank you! ❤️